core/trino-main/src/main/java/io/trino/sql/planner/planprinter/JsonRenderer.java

```diff
@@ -73,15 +75,20 @@ JsonRenderedNode renderJson(PlanRepresentation plan, NodeRepresentation node)
         private final List<JsonRenderedNode> children;
         private final List<String> remoteSources;
 
+        /**
+         * The @JsonCreator annotation is required to test explain query with JSON format.
+         * see: TestJsonRepresentation#testLogicalJsonPlan
+         */
+        @JsonCreator
```

Comment:
Why? We shouldn't have to, or even attempt to, deserialize those json objects to test it works. This class is meant to be a one way renderer, and not have the burden of dealing with deserialization.